Framework/Algorithms/src/ConvertToConstantL2.cpp

```diff
@@ -92,24 +89,22 @@ void ConvertToConstantL2::exec() {
     m_outputWS->setHistogram(i, m_inputWS->histogram(i));
 
     // Should not move the monitors
-    if (spectrumInfo.isMonitor(i)) {
+    if (inputSpecInfo.isMonitor(i))
       continue;
-    }
 
     // subract the diference in l2
-    double thisDetL2 = spectrumInfo.l2(i);
+    double thisDetL2 = inputSpecInfo.l2(i);
     double deltaL2 = std::abs(thisDetL2 - m_l2);
     double deltaTOF = calculateTOF(deltaL2);
     deltaTOF *= 1e6; // micro sec
 
     // position - set all detector distance to constant l2
     double r, theta, phi;
-    V3D oldPos = spectrumInfo.position(i);
+    V3D oldPos = inputSpecInfo.position(i);
     oldPos.getSpherical(r, theta, phi);
     V3D newPos;
     newPos.spherical(m_l2, theta, phi);
-    ComponentHelper::moveComponent(spectrumInfo.detector(i), pmap, newPos,
-                                   ComponentHelper::Absolute);
+    outputDetInfo.setPosition(i, newPos);
```

Comment:
I think having `DetectorGroup::getComponentID` throw should catch all of those situations so that you won't need the check? I agree that a new PR for that is what we want.